lib/bindings/window/window.go

```diff
@@ -27,9 +29,29 @@ type Window struct {
 	SendChannel      *connection.In `json:"-"`
 }
 
-func NewWindow(url string, sess *session.Session) *Window {
+func checkUrl(s string) (string, error) {
+	u, err := url.Parse(s)
+	if err != nil {
+		return s, err
+	}
+	if u.Scheme == "" {
+		p, err := filepath.Abs(s)
+		if err != nil {
+			return p, err
```

Comment:
Here you return the result of the failing call, but a few lines up you return the argument to the failing call. Why the difference?